### PR TITLE
update deps and minor cleanup

### PR DESCRIPTION
**Pull request by himanshuchawla009**
- Removed unused deps
- Removed  db.ts file from tss client and moved to example, as it was only used in example
- Fixed example
- Update torus-scripts to use latest build config
- Reordered code in client.ts file to meet linting criteria.
===================
Note
- Tested with prod example.

### CODE REVIEW

File: packages/tss-client/package.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-  "version": "3.0.0",
+  "version": "3.1.0",
   "description": "Client SDK for MPECDSA signing",
   "files": [
     "dist"
   ],
-  "main": "dist/tssClient.cjs.js",
-  "module": "dist/tssClient.esm.js",
+  "main": "dist/lib.cjs/index.js",
+  "module": "dist/lib.esm/index.js",
```
Comment (matthiasgeihs) [inline packages/tss-client/package.json:9]: for my understanding, why this change? the previously referenced files still exist.

File: packages/tss-client/src/client.ts [hunk 2]
```diff
@@ -473,3 +392,84 @@ export class Client {
...
+          sender: self_index,
+          recipient: party,
+          msg_type,
+          msg_data,
+        }),
+      });
+    }
+    return true;
+  };
+}
```
Comment (matthiasgeihs) [inline packages/tss-client/src/client.ts:475]: did you just move this piece of code? or also modify?

File: packages/tss-client/src/utils.ts [hunk 3]
```diff
@@ -112,10 +112,11 @@ export function getTSSPubKey(dkgPubKey: PointHex, userSharePubKey: key, userTSSI
   return serverTerm.add(userTerm);
 }
 
-export const generateEndpoints = (parties: number, clientIndex: number, tssEndpoints: string[], wsEndpoints: string[], nodeIndexes: number[]) => {
-  const endpoints = [];
-  const tssWSEndpoints = [];
-  const partyIndexes = [];
+export const generateEndpoints = (tssNodeEndpoints: string[], parties: number, clientIndex: number, nodeIndexes: number[]) => {
```
Comment (matthiasgeihs) [inline packages/tss-client/src/utils.ts:115]: can you outline the changes to this function? maybe a brief function documentation would be nice.

File: packages/tss-lib/torus.config.js [hunk 4]
```diff
@@ -1,4 +1,6 @@
 module.exports = {
   ...require("@toruslabs/config/torus.config"),
   esm: false,
+  libEsm: false,
+  libCjs: false,
```
Comment (matthiasgeihs) [inline packages/tss-lib/torus.config.js:5]: cjs is false but there is a config for cjs?

File: packages/tss-lib/webpack.config.js [hunk 5]
```diff
@@ -12,5 +12,5 @@ const cfg = {
 
 module.exports = {
   umdConfig: cfg,
-  cjsConfig: cfg,
+  cjsConfig: cfg
```
Comment (matthiasgeihs) [inline packages/tss-lib/webpack.config.js:15]: there is a config for cjs but `libCjs: false` in `torus.config.js`?

File: packages/web-example/src/prod.ts [hunk 6]
```diff
@@ -51,13 +50,13 @@ const runTest = async (testConfig: TestConfig) => {
   const verifier = "torus-test-health";
   torusSp.verifierName = verifier;
   torusSp.verifierId = testConfig.verifier_id || generateRandomEmail();
-
+  
   // init storage layer
-  const torusSL = new MockStorageLayer();
-
+  const torusSL =  new MockStorageLayer();
```
Comment (matthiasgeihs) [inline packages/web-example/src/prod.ts:55]: (nitpick) added space where there should be no space

File: packages/web-example/src/prod.ts [hunk 7]
```diff
@@ -87,29 +86,28 @@ const runTest = async (testConfig: TestConfig) => {
...
 
-  await tb.initialize({ useTSS: true, factorPub, deviceTSSShare, deviceTSSIndex }).catch((err) => {
+  await tb.initialize({ useTSS: true, factorPub, deviceTSSShare, deviceTSSIndex }).catch((err: unknown)=>{
     throw new Error(`${err} tkey_error`);
   });
   console.log("syncing tkey metadata");
 
-  await tb.syncLocalMetadataTransitions().catch((err) => {
+  await tb.syncLocalMetadataTransitions().catch((err: unknown)=>{
     throw new Error(`meta-data error:${err}`);
```
Comment (matthiasgeihs) [inline packages/web-example/src/prod.ts:99]: unrelated, but could add a space here
```suggestion
    throw new Error(`meta-data error: ${err}`);
```

File: packages/web-example/src/prod.ts [hunk 8]
```diff
@@ -87,29 +86,28 @@ const runTest = async (testConfig: TestConfig) => {
...
   });
 
   const tssNonce = tb.metadata.tssNonces[tb.tssTag];
   console.log("fetching tss keys");
 
-
-  const { pubKey: pubKeyDetails } = await tb.serviceProvider.getTSSPubKey(tb.tssTag, tssNonce).catch((err) => {
+  
+  const { pubKey: pubKeyDetails } = await tb.serviceProvider.getTSSPubKey(tb.tssTag, tssNonce).catch((err)=>{
```
Comment (matthiasgeihs) [inline packages/web-example/src/prod.ts:106]: should be spaces around `=>`. i assume linter/prettier is not running for test code?
```suggestion
  const { pubKey: pubKeyDetails } = await tb.serviceProvider.getTSSPubKey(tb.tssTag, tssNonce).catch((err) => {
```

File: packages/web-example/src/prod.ts [hunk 9]
```diff
@@ -134,9 +133,9 @@ const runTest = async (testConfig: TestConfig) => {
 
   // 4. derive tss pub key, tss pubkey is implicitly formed using the dkgPubKey and the userShare (as well as userTSSIndex)
   const tssPubKey = getTSSPubKey(dkgTssPubKey, userSharePubKey, userTSSIndex);
-  const pubKey = Buffer.from(`${tssPubKey.getX().toString(16, 64)}${tssPubKey.getY().toString(16, 64)}`, "hex").toString("base64");
+  const pubKey = Buffer.from(`${tssPubKey.getX().toString(16, 64)}${tssPubKey.getY().toString(16,64)}`, "hex").toString("base64");
```
Comment (matthiasgeihs) [inline packages/web-example/src/prod.ts:136]: why remove space?
```suggestion
  const pubKey = Buffer.from(`${tssPubKey.getX().toString(16, 64)}${tssPubKey.getY().toString(16, 64)}`, "hex").toString("base64");
```

File: packages/web-example/src/prod.ts [hunk 10]
```diff
@@ -134,9 +133,9 @@ const runTest = async (testConfig: TestConfig) => {
...
 
-  // 5. Derive coeffs to make user share additive.
+    // 5. Derive coeffs to make user share additive.
```
Comment (matthiasgeihs) [inline packages/web-example/src/prod.ts:138]: why add space?

File: packages/web-example/src/prod.ts [hunk 11]
```diff
@@ -151,19 +150,24 @@ const runTest = async (testConfig: TestConfig) => {
...
+  client.log = (...args: unknown[]) => {
+    log(testConfig.label, ...args);
+  };
   console.log("doing tss precompute");
 
   // initiate precompute
   client.precompute({ signatures, server_coeffs: serverCoeffs });
-  await client.ready().catch((err) => {
-    throw new Error(`tss-error during compute: ${err}`);
+  await client.ready().catch((err)=>{
```
Comment (matthiasgeihs) [inline packages/web-example/src/prod.ts:163]: missing space

File: packages/web-example/src/prod.ts [hunk 12]
```diff
@@ -151,19 +150,24 @@ const runTest = async (testConfig: TestConfig) => {
...
+    throw new Error(`tss_error during compute: ${err}`);
   });
   console.log("ready for signing");
 
   // initiate signature.
-  const signature = await client.sign(msgHash.toString("base64"), true, msg, "keccak256", { signatures }).catch((err) => {
-    throw new Error(`${testConfig.label} error during sign: ${err}`);
+  const signature = await client.sign(msgHash.toString("base64"), true, msg, "keccak256", { signatures }).catch((err)=>{
```
Comment (matthiasgeihs) [inline packages/web-example/src/prod.ts:169]: missing space

**Review by matthiasgeihs — COMMENTED** [review]
there is a bunch of weird spacing issues in `web-example/src/prod.ts`. what's up with that?

File: packages/tss-client/package.json (hunk 1 above)
Comment (himanshuchawla009) [reply packages/tss-client/package.json:9]: its some build optimization chai has done in torus-scripts, new files are more tree shakeable. @chaitanyapotti can share more about those optimizations.

File: packages/tss-client/src/client.ts (hunk 2 above)
Comment (himanshuchawla009) [reply packages/tss-client/src/client.ts:475]: just moved, eslint was complaining because it was being used before its defined

File: packages/tss-client/src/utils.ts (hunk 3 above)
Comment (himanshuchawla009) [reply packages/tss-client/src/utils.ts:115]: previous function was not being used by any apps or sdks because it was hardcoding urls and was only working for sapphire mainnet, so whats happening now that all the sdks and examples ends up implementing their own generateEndpoints function. So in order to avoid the code duplication i changed this function to become exactly same as mpc core kit one as implemented here: https://github.com/Web3Auth/mpc-core-kit/blob/2ae3b41d816a94aabd013c0ceffaf74937e9803d/src/utils.ts#L41

File: packages/tss-lib/torus.config.js (hunk 4 above)
Comment (himanshuchawla009) [reply packages/tss-lib/torus.config.js:5]: libCjs and cjs are now 2 different formats exposed by torus-scripts, cjs still stays, i havent removed that.  libCjs  basically creates build using rollup instead of webpack but it doesnt works with wasm currently so i decided to stick with cjs with webpack only.

File: packages/tss-lib/webpack.config.js (hunk 5 above)
Comment (himanshuchawla009) [reply packages/tss-lib/webpack.config.js:15]: replied above

File: packages/tss-client/src/utils.ts (hunk 3 above)
Comment (metalurgical) [reply packages/tss-client/src/utils.ts:115]: This function had hard-coded values in it. #70 corrected this, though it has changed again here. Lets settle this tomorrow.

**Review by matthiasgeihs — APPROVED** [review]
